app/components/goodcity/locations-search-overlay.js

```diff
@@ -51,11 +68,22 @@ export default SingletonComponent.extend(SearchMixin, {
     },
 
     loadMoreLocations(pageNo) {
-      const params = this.trimQuery(
-        _.merge({}, this.getSearchQuery(), this.getPaginationQuery(pageNo))
-      );
+      if (!this.isValidTextLength()) {
+        return;
+      }
+
+      if (this.get("presetLocations.length")) {
+        let searchText = this.getSearchQuery().searchText.toLowerCase();
+        return this.get("presetLocations").filter(
+          this.isLocationPresent(searchText)
```

Comment:
I think it would be better to access `searchText` in `isLocationPresent` method itself 🤔  instead of passing for every location.